lib/sync.js

```diff
@@ -41,9 +41,8 @@ function pbkdf2 (password, salt, iterations, keylen, digest) {
       for (var k = 0; k < hLen; k++) T[k] ^= U[k]
     }
 
-    var destPos = (i - 1) * hLen
-    var len = (i === l ? r : hLen)
-    T.copy(DK, destPos, 0, len)
```

Comment:
This is perfectly fine.

`r` is just the remainder to avoid a Buffer overrun.
Check by seeing that `r == keylen % hLen`.... bah of course it is literally the `remainder`.